op-e2e/actions/interop/proofs_test.go

```diff
@@ -450,6 +450,87 @@ func TestInteropFaultProofs_CascadeInvalidBlock(gt *testing.T) {
 	runFppAndChallengerTests(gt, system, tests)
 }
 
+func TestInteropFaultProofs_MessageExpiry(gt *testing.T) {
+	t := helpers.NewDefaultTesting(gt)
+	// TODO(#14234): Check message expiry in op-supervisor
+	t.Skip("Message expiry not yet implemented")
+
+	system := dsl.NewInteropDSL(t)
+
+	actors := system.Actors
+	alice := system.CreateUser()
+	emitterContract := dsl.NewEmitterContract(t)
+	system.AddL2Block(actors.ChainA, dsl.WithL2BlockTransactions(
+		emitterContract.Deploy(alice),
+	))
+	system.AddL2Block(actors.ChainA, dsl.WithL2BlockTransactions(
+		emitterContract.EmitMessage(alice, "test message"),
+	))
+	emitTx := emitterContract.LastEmittedMessage()
+
+	// Bring ChainB to the same height and timestamp
+	system.AddL2Block(actors.ChainB)
+	system.AddL2Block(actors.ChainB)
+	system.SubmitBatchData()
+
+	// Advance the chain until the init msg expires
+	msgExpiryTime := actors.ChainA.RollupCfg.GetMessageExpiryTimeInterop()
+	end := emitTx.Identifier().Timestamp.Uint64() + msgExpiryTime
+	for {
+		if system.Actors.ChainB.Sequencer.L2Unsafe().Time >= end {
+			break
+		}
+		system.AddL2Block(actors.ChainA)
+		system.AddL2Block(actors.ChainB)
+	}
```

Comment:
I'd suggest pushing this logic into the DSL since repeating blocks until a certain timestamp is a common pattern: https://github.com/ethereum-optimism/optimism/pull/14368

Reply:
Thanks for the patch. Merged.